src/codec.rs

```diff
@@ -82,19 +78,52 @@ impl From<Utf8Error> for ParseError {
 /// Encodes and decodes Language Server Protocol messages.
 #[derive(Clone, Debug)]
 pub struct LanguageServerCodec<T> {
-    remaining_msg_bytes: usize,
+    http_error: Option<httparse::Error>,
+    headers_len: Option<usize>,
+    content_len: Option<usize>,
     _marker: PhantomData<T>,
 }
 
+impl<T> LanguageServerCodec<T> {
+    fn reset(&mut self) {
+        self.http_error = None;
+        self.headers_len = None;
+        self.content_len = None;
+    }
+}
+
 impl<T> Default for LanguageServerCodec<T> {
     fn default() -> Self {
         LanguageServerCodec {
-            remaining_msg_bytes: 0,
+            http_error: None,
+            headers_len: None,
+            content_len: None,
             _marker: PhantomData,
         }
     }
 }
 
+#[cfg(feature = "runtime-agnostic")]
+impl<T: Serialize> Encoder for LanguageServerCodec<T> {
+    type Error = ParseError;
+    type Item = T;
+
+    fn encode(&mut self, item: Self::Item, dst: &mut BytesMut) -> Result<(), Self::Error> {
+        let msg = serde_json::to_string(&item)?;
+        log::trace!("-> {}", msg);
```

Comment:
Since `trace!()` is already imported, we might want to remove the `log::` prefix to eliminate warnings.